azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -398,12 +402,78 @@ public void createContainer(final int executionId) throws ExecutorManagerExcepti
    */
   @Override
   public void deleteContainer(final int executionId) throws ExecutorManagerException {
-    deletePod(executionId);
+    try { // if pod deletion is not successful, the service deletion can still be handled
+      deletePod(executionId);
+    } catch (ExecutorManagerException e) {
+      logger.error("ExecId: {}, Unable to delete Pod in Kubernetes: {}", executionId,
+          e.getMessage());
+    }
     if (isServiceRequired()) {
       deleteService(executionId);
     }
   }
 
+  /**
+   * This method is used to delete invalid containers in batch
+   * @param containerValidity pod validity in milliseconds
+   * @throws ExecutorManagerException
+   */
+  @Override
+  public void deleteContainers(long containerValidity) throws ExecutorManagerException {
+    logger.info(String.format("Cleaning up containers older than %d days",
+        containerValidity / DAY_TO_MILLISECOND_CONVERSION_FACTOR));
+    final List<String> podsToDelete = getStalePods(containerValidity);
+    for (String podName : podsToDelete) {
+      logger.info(String.format("Deleting stale pod %s", podName));
+      final int execId = getExecutionId(podName);
+      if (execId > 0) {
+        deleteContainer(execId);
```

Comment:
Seems like every webserver will be deleting all the stale pods including those for other Azkaban clusters if they happen to be sharing the same kubernetes namespace. 
Is this intentional?

Reply:
Thank you for reviewing. In ln 443, the label indicates that only pods belong to current az cluster is queried (final String label = "cluster=" + this.clusterName;).